### ISSUE TITLE
The `@stylistic/less/unit-case` rule recases what a hyphen welds to a unit, which Less reads as a subtraction

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#633]
### What minimal example or steps are needed to reproduce the bug?

A dimension a hyphen welds to a word Less compiles as an operand of its own:

```less
a { width: 10PX-A; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "overrides": [
    {
      "files": ["**/*.less"],
      "customSyntax": "postcss-less",
      "rules": {
        "@stylistic/less/unit-case": "lower"
      }
    }
  ]
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json
```

### What did you expect to happen?

Nothing said about `A`. Less reads a hyphen behind a dimension as the sign of the operand behind it: `less@4.9.1` compiles that line to `a { width: 10PX A; }`, a dimension whose unit is `PX` and a keyword it prints exactly as it was written, and `a { width: 10PX-2REM; }` to `a { width: 8PX; }`. So `-A` is no part of the unit under this syntax, whatever it is under plain CSS. A hyphen is one of the characters that keep Less from storing a declaration's value unread, so this holds in every place Less reads a value — a variable's value and a set of media parameters among them.

### What actually happened?

```
1:12-16  Expected "PX-A" to be "px-a"
```

and `--fix` leaves `a { width: 10px-a; }`, which Less compiles to `a { width: 10px a; }` — the keyword recased along with the unit. In a variable's value the same: `@v: 10PX-A;` is reported at `1:7-11` and comes out as `@v: 10px-a;`.

Measured on `71543a3`, the branch of [#527](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/527), through the plugin with `postcss-less`.

### What is the cause?

The reading is plain CSS's, and it is right there: a hyphen is a code point of an identifier, so `@csstools/css-tokenizer` reads `10PX-A` as one dimension whose unit is `PX-A`, and `lightningcss` prints the word whole. `@stylistic/less/unit-case` inherits that reading. Less reads a unit as ASCII letters and underscores alone (`entities.dimension` in `parser.js`, `^([+-]?\d*\.?\d+)(%|[a-z_]+)?`), so its unit ends in front of the hyphen and the hyphen opens a subtraction.

The same reader ends a unit at a digit (`10PX9` compiles to `10PX 9`) and reads no exponent (`1E5PX` compiles to `1E 5PX`), where no case is at stake and the rule's write is what Less would want anyway; and at an escape, which is #527.

### Do you have a proposal to fix the bug?

Widen the reading #527 gives the Less syntax — the unit ends in front of the first escape — to the hyphen, so the unit ends where Less ends it there too. `10PX-2REM` under `lower` is then two units to name, `PX` and `REM`, the way `10PX*2REM` is, and both are written.
